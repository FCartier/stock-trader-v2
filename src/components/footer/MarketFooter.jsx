import React from "react";

const MarketFooter = props => {
  const { market } = props;
  
  return (
    <div>
      <ul className="market-foot">
        {market &&
          market.map(item => <li key={item.mic}><p><strong>{item.venueName}</strong>  {item.marketPercent}%</p></li>)}
      </ul>
    </div>
  );
};

export default MarketFooter;
