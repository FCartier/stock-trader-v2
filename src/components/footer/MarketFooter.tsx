import React from "react";

interface IMarketItem {
  mic: string;
  venueName: string;
  marketPercent: number;
}

interface IMarketsProps {
  market?: [];
}

const MarketFooter: React.SFC<IMarketsProps> = ({market}) => {
  return market.length ? (
    <div>
      <ul className="market-foot">
        {market.map((item: IMarketItem, index: number)  => <li key={index}><p><strong>{item.venueName}</strong>  {item.marketPercent}%</p></li>)}
      </ul>
    </div>
  ) : null;
};

export default MarketFooter;