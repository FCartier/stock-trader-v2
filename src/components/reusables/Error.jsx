import React from 'react'

const Error = props => {
    const { error } = props
    return (
        <div>
            {error && 
             <h2>Error, try another search</h2>
            }
        </div>
    )
}

export default Error;