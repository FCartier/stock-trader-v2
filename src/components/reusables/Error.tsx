import React from 'react';

interface ErrorProps {
    error : {
        message: string;
        stack: string;
    }
}

const Error: React.SFC<ErrorProps> = ({error}) => {
    return !(typeof error === "undefined" || error.message === "") ? (
        <div><h2>Error, try another search</h2></div>
    ) : null;
}

export default Error;