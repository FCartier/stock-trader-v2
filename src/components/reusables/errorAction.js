export const ERROR_ACTION = "ERROR ACTION";

export function errorAction(payload) {
  return {
    type: ERROR_ACTION,
    payload
  };
}