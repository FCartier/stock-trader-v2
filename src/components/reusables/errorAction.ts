import { Action, ActionCreator } from 'redux';

export const ERROR_ACTION = "ERROR_ACTION";
export type ERROR_ACTION = typeof ERROR_ACTION;

export interface ErrorState {
    message: string;
    stack: string;
}

export interface ErrorAction extends Action {
  type: typeof ERROR_ACTION
  error: ErrorState
}

export const errorAction: ActionCreator<ErrorAction> = (error: ErrorState) => ({
  type: ERROR_ACTION,
  error
})
