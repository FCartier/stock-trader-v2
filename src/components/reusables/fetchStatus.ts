export enum fetchStatus {
  pending,
  success,
  failed
}