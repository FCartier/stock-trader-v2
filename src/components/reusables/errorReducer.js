import { ERROR_ACTION } from "./errorAction"

export const error = (state = {}, action) => {
    switch (action.type) {
      case ERROR_ACTION:
        return action.payload;
      default:
        return state;
    }
  };