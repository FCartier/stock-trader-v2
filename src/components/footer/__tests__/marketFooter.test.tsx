import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import MarketFooter from "../MarketFooter";
import marketData from "../__data__/marketFooter.data";
import { fetchStatus } from '../../reusables/fetchStatus';

configure({ adapter: new Adapter() });

describe("<MarketFooter />", () => {
  test("should render properly", () => {
    const component = shallow(MarketFooter({ market: marketData, status: fetchStatus.success }));
    expect(component).toMatchSnapshot();
  });
});