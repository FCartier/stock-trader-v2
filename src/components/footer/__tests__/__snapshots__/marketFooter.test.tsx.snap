// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarketFooter /> should render properly 1`] = `
<ul
  className="css-k5mi76"
>
  <li
    key="0"
  >
    <p>
      <strong>
        TRF Volume
      </strong>
       
      0.37027
      %
    </p>
  </li>
  <li
    key="1"
  >
    <p>
      <strong>
        NASDAQ
      </strong>
       
      0.13443
      %
    </p>
  </li>
  <li
    key="2"
  >
    <p>
      <strong>
        NYSE
      </strong>
       
      0.12838
      %
    </p>
  </li>
  <li
    key="3"
  >
    <p>
      <strong>
        NYSE Arca
      </strong>
       
      0.11331
      %
    </p>
  </li>
</ul>
`;
